test/ttmlir/Dialect/TTNN/ccl/all_gather.mlir

```diff
@@ -0,0 +1,11 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s | FileCheck %s
```

Comment:
Can you add a negative test case as well? To test the verify failure for the dim attribute?